core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -1257,7 +1328,7 @@ class LogTest {
     log.close()
 
     // After reloading log, producer state should not be regenerated
-    val reloadedLog = createLog(logDir, logConfig, logStartOffset = 1L)
+    val reloadedLog = createLog(logDir, logConfig, logStartOffset = 1L, lastShutdownClean = false)
```

Comment:
Jun, this test was not creating a clean shut down file before opening the log again. So, it would have been going through log recovery code path earlier as well.